config/config_mgmt.py

```diff
@@ -253,7 +253,77 @@ def __init__(self, source="configDB", debug=False, allowTablesWithoutYang=True):
     def __del__(self):
         pass
 
-    def _checkKeyinAsicDB(self, key, db):
+    def _checkPortsDownAppDb(self, db, ports):
+        '''
+        Check APP DB for PORT oper status down.
+
+        Parameters:
+            db (SonicV2Connector): database.
+            ports (list): List of ports
+
+        Returns:
+            (bool): True, if all ports are oper down.
+        '''
+        try:
+            # connect to APP DB,
+            db.connect(db.APPL_DB)
+            for port in ports:
+                # Key format "PORT_TABLE:Ethernet112"
+                key = 'PORT_TABLE:{}'.format(port)
+                portOperStatus = db.get('APPL_DB', key, 'oper_status')
+                self.sysLog(syslog.LOG_DEBUG, "APPL_DB: {} {}".\
+                    format(key, portOperStatus))
+                if portOperStatus is None or portOperStatus != 'down':
+                    return False
+
+        except Exception as e:
+            self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
+            return False
+
+        return True
+
+    def _verifyAppDB(self, db, ports, timeout):
+        '''
+        Verify in the App DB that ports are oper_status down,
+        Keep on trying till timeout period.
+
+        Parameters:
+            db (SonicV2Connector): database.
+            ports (list): List of ports
+            timeout (int): timeout period.
+
+        Returns:
+            (bool): True, if all ports are oper down.
+        '''
+        self.sysLog(msg="Verify Port oper_status down from App DB, Wait...")
+
+        try:
+            for waitTime in range(timeout):
+                self.sysLog(logLevel=syslog.LOG_DEBUG, \
+                    msg='Check App DB: {} try'.format(waitTime+1))
+                # checkPortsDownAppDb will return True if all ports are Oper
+                # down in APP DB
+                if self._checkPortsDownAppDb(db, ports):
+                    break
+                tsleep(1)
```

Comment:
>tsleep [](http://example.com/codeflow?start=16&length=6)

@daall, Do we  have better way to wait a hash field in Redis?

Reply:
@qiluo-msft 
that will be good, Note: we need to wait till the value of the field becomes a specific value i.e. "down" here or till timeout.

Reply:
I believe we have a "wait_until" concept in sonic-mgmt, but I don't think there is an analogue to that in the image currently. Is that something we should consider adding to swss-common, @qiluo-msft ?

Reply:
Could you share a sample implementation?